app/furniture/marketplace/routes.rb

```diff
@@ -9,8 +9,12 @@ def self.append_routes(router)
           router.resource :checkout, only: [:show, :create]
         end
 
+        router.resources :tax_rates
+
         router.resources :orders, only: [:show]
-        router.resources :products
+        router.resources :products do
+          router.resources :tax_rates, only: [:create, :destroy, :index, :show]
```

Comment:
I only see code to edit and display the Marketplace Tax Rates, but there is this `products/tax_rates` set of routes, and also a reference from the marketplace products table to tax rates, although the `TaxRate` model merely `belongs_to :marketplace` -- maybe you are missing the `has_many :marketplace_products` on `TaxRate` and the `belongs_to :tax_rate, optional: true` on `MarketplaceProduct`? 

I'm not sure if the Tax Rates on Products are left over from a previous version, or if you intend to use those somehow differently?

Reply:
I should get rid of this since I haven't started the associating `TaxRate`s to `Product`s yet.